CRM/Contribute/BAO/Contribution.php

```diff
@@ -3549,6 +3564,7 @@ public static function updateFinancialAccounts(&$params, $context = NULL, $skipT
       }
     }
     $trxn = CRM_Core_BAO_FinancialTrxn::create($params['trxnParams']);
+    $previousLineItem = CRM_Price_BAO_LineItem::getLineItemsByContributionID($params['contribution']->id);
```

Comment:
so the variable name is $previousLineItem but you are passing current contribution id .. should you pass $params['prevContribution']->id instead .. so 
```php
previousLineItem = CRM_Price_BAO_LineItem::getLineItemsByContributionID($params['prevContribution']->id);
```

in case is what you want is the current contribution line item then I think you should rename the variable name to reflect that .. or maybe I am missing something again ?